src/config/configresolver.cpp

```diff
@@ -0,0 +1,133 @@
+#include "src/config/configresolver.h"
+#include "src/config/configerrordetails.h"
+#include "src/utils/confighandler.h"
+
+#include "src/utils/valuehandler.h"
+#include <QDialogButtonBox>
+#include <QLabel>
+#include <QVBoxLayout>
+
+ConfigResolver::ConfigResolver(QWidget* parent)
+  : QDialog(parent)
+{
+    setWindowTitle("Resolve configuration errors");
+    setMinimumSize({ 250, 200 });
+    setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Maximum);
+    populate();
+    connect(ConfigHandler::getInstance(),
+            &ConfigHandler::fileChanged,
+            this,
+            [this]() { populate(); });
+}
+
+QGridLayout* ConfigResolver::layout()
+{
+    return dynamic_cast<QGridLayout*>(QDialog::layout());
+}
+
+void ConfigResolver::populate()
+{
+    ConfigHandler config;
+    QList<QString> unrecognized;
+    QList<QString> semanticallyWrong;
+
+    config.checkUnrecognizedSettings(nullptr, &unrecognized);
+    config.checkSemantics(nullptr, &semanticallyWrong);
+
+    // Remove previous layout and children, if any
+    resetLayout();
+
+    bool anyErrors = !semanticallyWrong.isEmpty() || !unrecognized.isEmpty();
+    int i = 0;
```

Comment:
It took me a minute to figure out what 'i' represents here. Can you rename it something descriptive like widgetPos ?